### PR TITLE
fix dns controller crashing for DO

### PR DESCRIPTION
**Pull request by zak905**
More details can be found in the issue #13441

### CODE REVIEW

**Comment by justinsb** [comment]
Thanks for tracking down and fixing.  Is making the DO provider accept multiple records an option?  That might avoid encoding the package name as a string (without introducing a circular package ref)

**Comment by zak905** [comment]
you are welcome. I have already made an attempt here https://github.com/zak905/kops/commit/40136a0c8eceb088c58379c5dd941f876eeb6781 to make the DO DNS provider accomodate multiple values.

I have not tested it yet in my cluster, but if you confirm this is a better alternative, I can test it and update the PR

**Comment by zak905** [comment]
I have just tested now the changes in https://github.com/zak905/kops/commit/40136a0c8eceb088c58379c5dd941f876eeb6781 
They seem to work well, no crashes, also the records seem to have the correct values. 
I am updating this PR in this case, if this is the prefeffered approach.

File: dnsprovider/pkg/dnsprovider/providers/do/dns.go [hunk 1]
```diff
@@ -185,20 +185,27 @@ func (r *resourceRecordSets) List() ([]dnsprovider.ResourceRecordSet, error) {
 		return nil, err
 	}
 
-	var rrset *resourceRecordSet
 	var rrsets []dnsprovider.ResourceRecordSet
+	rrsetWithoutDups := make(map[string]*resourceRecordSet)
```
Comment (hakman) [inline dnsprovider/pkg/dnsprovider/providers/do/dns.go:189]: Should this be plural to keep similar with `rrsets` above?
```suggestion
	rrsetsWithoutDups := make(map[string]*resourceRecordSet)
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: dnsprovider/pkg/dnsprovider/providers/do/dns.go (hunk 1 above)
Comment (zak905) [reply dnsprovider/pkg/dnsprovider/providers/do/dns.go:189]: sure

Comment (hakman) [reply dnsprovider/pkg/dnsprovider/providers/do/dns.go]: Thanks. Please also squash the commits and change the commit message to something explaining the change, like "[DigitalOcean]: accept multiple records in dns-controller".
/lgtm

Comment (zak905) [reply dnsprovider/pkg/dnsprovider/providers/do/dns.go]: Done.

Comment (hakman) [reply dnsprovider/pkg/dnsprovider/providers/do/dns.go]: 👍

**Review by hakman — APPROVED** [review]

**Pull request merged**